src/SingleCellAnnotator.imjoy.html

```diff
@@ -406,56 +418,90 @@
     },
 
     async uploadAnnotation(){
-      // Default options are marked with *
-      const sendData = JSON.parse(JSON.stringify(sendDataTemplate));
-      sendData.discarded = this.current_image_data.discarded;
-      if (this.current_image_data.discarded){
-        // delete sendData.mask;
-        sendData.mask = null;
+      // skip upload annotation if image is annotated and no new change
+      if (this.current_image_data.annotation_status&&(this.current_image_data.mask_adjusted===null)&&(this.current_image_data.location_adjusted===null)){
+        // skip uploadannotation
+        return
       }else{
-        sendData.bbox = this.current_image_data.mask_geojson.bbox;
-        sendData.location = {};
-        if (this.current_image_data.label_display_mode == 'single'){
-          sendData.location[this.current_image_data.location] = this.current_location_selected;
+        // Default options are marked with *
+        const sendData = JSON.parse(JSON.stringify(sendDataTemplate));
+        sendData.discarded = this.current_image_data.discarded;
+        if (this.current_image_data.discarded){
+          //delete sendData.mask;
+          sendData.mask = null;
+        }else{
+          sendData.bbox = this.current_image_data.mask_geojson.bbox;
+          sendData.location = {};
+          if (this.current_image_data.label_display_mode == 'single'){
+            // do not upload annotion if no location selected
+            if (this.current_location_selected===null){
+              if (this.current_image_data.mask_adjusted===null){
+                return
+              }else{
+              const anwser = await api.confirm("Please selecte Yes/No to annotate?")
+              this.current_image_data.annotation_status = null;
+              return
+              };
+            }else{
+              // upload the re-selection
+              this.current_image_data.annotation_status = false;
+            }
+            sendData.location[this.current_image_data.location] = this.current_location_selected;
           } else if (this.current_image_data.label_display_mode == 'multiple'){
             sendData.location = this.current_image_data.location;
             for (loc in sendData.location){
-              if(sendData.location[loc]==null){
-                sendData.location[loc] = false}
-            }
+              if((sendData.location[loc]===null)||(sendData.location[loc]===false)){
+                sendData.location[loc] = false
+              }else{
+                this.current_image_data.annotation_status = false}}
+            if (this.current_image_data.annotation_status===null){
```

Comment:
```suggestion
            if (!this.current_image_data.annotation_status){
```

Reply:
Not correct. I have this.current_location_selected===false. which is different from null

Reply:
Not correct. I have this.current_image_data.annotation_status===false. which is different from null

Reply:
It's a bad practice to use both `false` and `null` to represent different meaning, you should use other value instead.

Reply:
I will make another PR to fix that